git/init.go

```diff
@@ -111,5 +112,38 @@ func Init(c *Client, opts InitOptions, dir string) (*Client, error) {
 	}
 	c.WorkDir = WorkDir(wd)
 	c.GitDir = GitDir(wd + "/.git")
+
+	if opts.Template.String() != "" {
+		err = filepath.Walk(opts.Template.String(), func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			path, err = filepath.Rel(opts.Template.String(), path)
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				os.MkdirAll(filepath.Join(c.GitDir.String(), path), 0777)
+			} else {
+				newFile, err := os.Create(filepath.Join(c.GitDir.String(), path))
```

Comment:
The `git.GitDir` type has a create helper method intended to make this simpler. You could just do

```
newfile, err := c.GitDir.Create(File(path));
```

to create a file named path under the Client's GitDir.